docs: translation for useRef (#356)
reactjs/id.react.dev: src/content/reference/react/useRef.md

  // ...
```

[Lihat lebih banyak contoh di bawah ini.](#usage)

#### Parameters {/*parameters*/}

* `initialValue`: Nilai yang Anda inginkan untuk properti `current` objek ref pada awalnya. Nilai ini dapat berupa nilai dari jenis apa pun. Argumen ini akan diabaikan setelah render awal.

#### Returns {/*returns*/}

`useRef` mengembalikan sebuah objek dengan satu properti:

> * `current`: Awalnya, ini diatur ke `initialValue` yang telah Anda lewati. Anda dapat mengaturnya ke nilai yang lain. Jika Anda mengoper objek ref ke React sebagai sebuah atribut `ref` ke node JSX, React akan menyetel properti `current`.

Reviewer: ```suggestion
* `current`: Awalnya, ini diatur ke `initialValue` yang telah Anda oper. Anda dapat mengaturnya ke nilai yang lain. Jika Anda mengoper objek ref ke React sebagai sebuah atribut `ref` ke node JSX, React akan menyetelkan properti `current`-nya.
```

Author: Thanks atas koreksi nya @mazipan , kelupaan passed itu ada yang emang sudah oper dan ini terlewat :D

#### Perhatian {/*caveats*/}

* Anda dapat mengubah properti `ref.current`. Tidak seperti state, properti ini tidak dapat diubah. Namun, jika properti ini menyimpan objek yang digunakan untuk rendering (misalnya, sebuah bagian dari state Anda), maka Anda tidak boleh mengubah objek tersebut.
* Ketika Anda mengubah properti `ref.current`, React tidak me-render ulang komponen Anda. React tidak mengetahui kapan Anda mengubahnya karena ref adalah objek JavaScript biasa.
* Jangan menulis _atau membaca_ `ref.current` selama proses rendering, kecuali untuk [inisialisasi.](#avoiding-recreating-the-ref-contents) Hal ini membuat perilaku komponen Anda tidak dapat diprediksi.
* Dalam Strict Mode, React akan **memanggil fungsi komponen Anda dua kali** untuk [membantu Anda menemukan bug yang tidak disengaja.](#my-initializer-or-updater-function-runs-twice) Ini adalah perilaku khusus development dan tidak mempengaruhi produksi. Setiap objek ref akan dibuat dua kali, tetapi salah satu versi akan dibuang. Jika fungsi komponen Anda murni (sebagaimana mestinya), hal ini seharusnya tidak mempengaruhi perilaku.

---

## Penggunaan {/*usage*/}

> ### Merujuk sebuah nilai dengan sebuah ref {/*referencing-a-value-with-a-ref*/}

Reviewer: Di atas ada subtitle `Reference` yang diterjemahkan dengan `Referensi`. 

Meskipun menurutku di sini `Merujuk` juga masih cocok saja dengan konteks yang diinginkan, tapi kalau mau konsisten sepertinya lebih baik pakai kata yang sama.

```suggestion
### Mereferensikan sebuah nilai dengan sebuah ref {/*referencing-a-value-with-a-ref*/}
```

Reviewer 2: Betul, lebih pas `Mereferensikan`

}
```

> Nantinya, Anda dapat membaca ID interval tersebut dari ref sehingga Anda dapat memanggil [clear that interval](https://developer.mozilla.org/en-US/docs/Web/API/clearInterval):

Reviewer: ```suggestion
Nantinya, Anda dapat membaca ID interval tersebut dari ref sehingga Anda dapat memanggil [pembersihan interval tersebut](https://developer.mozilla.org/en-US/docs/Web/API/clearInterval):
```

}
```

Dengan menggunakan ref, Anda memastikan bahwa:

- Anda dapat **menyimpan informasi** di antara render ulang (tidak seperti variabel biasa, yang disetel ulang pada setiap render).
- Mengubahnya **tidak memicu sebuah render ulang** (tidak seperti variabel state, yang memicu sebuah render ulang).

> - The **information is local** to each copy of your component (unlike the variables outside, which are shared).

Reviewer: Kurang bagian ini, kelewatan belum diterjemahkan:

```
- The **information is local** to each copy of your component (unlike the variables outside, which are shared).
```

Reviewer 2: ```suggestion
- **Informasinya lokal** untuk setiap salinan komponen Anda (tidak seperti variabel di luar, yang dibagikan).
```

Reviewer 2: @xevorte It's okay too, `Informasi ini bersifat lokal` should be fine.

Prior I suggested that because the source doesn't have `this` word. It wasn't `This **information is local**` but rather `The **information is local**`.

<Pitfall>

**Jangan menulis _atau membaca_ `ref.current` selama proses rendering.**

> React mengharapkan bahwa tubuh komponen Anda [berperilaku seperti pure function](/learn/keeping-components-pure):

Reviewer: ```suggestion
React mengharapkan bahwa tubuh komponen Anda [berperilaku seperti fungsi yang murni](/learn/keeping-components-pure):
```

  // ...
```

Kemudian oper objek ref Anda sebagai atribut `ref` ke JSX milik node DOM yang ingin Anda manipulasi:

```js [[1, 2, "inputRef"]]
  // ...
  return <input ref={inputRef} />;
```

> Setelah React membuat node DOM dan meletakkannya di layar, React akan mengatur <CodeStep step={2}>properti `current`</CodeStep> objek Anda yang merujuk ke node DOM tersebut. Sekarang Anda dapat mengakses `<input>`'s node DOM dan method panggilan seperti [`focus()`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/focus):

Reviewer: ```suggestion
Setelah React membuat node DOM dan meletakkannya di layar, React akan mengatur <CodeStep step={2}>properti `current`</CodeStep> objek Anda yang merujuk ke node DOM tersebut. Sekarang Anda dapat mengakses node DOM dari `<input>` dan memanggil method seperti [`focus()`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/focus):
```

Reviewer 2: @mazipan @resir014 `method` or translated into `metode`?

<Solution />

#### Menggulir gambar ke dalam tampilan {/*scrolling-an-image-into-view*/}

> Pada contoh ini, mengeklik tombol akan menggulir gambar ke dalam tampilan. Ini menggunakan ref ke daftar node DOM, dan kemudian memanggil DOM [`querySelectorAll`](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll) API untuk menemukan gambar yang ingin kita gulir.

Reviewer: ```suggestion
Pada contoh ini, mengeklik tombol akan menggulirkan gambar ke dalam tampilan. Ini menggunakan ref ke daftar node DOM, dan kemudian memanggil DOM [`querySelectorAll`](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll) API untuk menemukan gambar yang ingin kita gulir.
```

<Solution />

#### Mengekspos sebuah ref ke komponen Anda sendiri {/*exposing-a-ref-to-your-own-component*/}

> Terkadang, Anda mungkin ingin membiarkan komponen induk memanipulasi DOM di dalam komponen Anda. Sebagai contoh, mungkin Anda menulis komponen `MyInput`, tetapi Anda ingin induknya dapat memfokuskan masukan (yang tidak dapat diakses oleh induknya). Anda dapat menggunakan kombinasi `useRef` untuk menampung masukan dan [`forwardRef`](/reference/react/forwardRef) untuk mengeksposnya ke komponen induk. Membaca sebuah [panduan mendetail](/learn/manipulating-the-dom-with-refs#accessing-another-components-dom-nodes) disini.

Reviewer: ```suggestion
Terkadang, Anda mungkin ingin membiarkan komponen induk memanipulasi DOM di dalam komponen Anda. Sebagai contoh, mungkin Anda menulis komponen `MyInput`, tetapi Anda ingin induknya dapat memfokuskan masukan (yang tidak dapat diakses oleh induknya). Anda dapat menggunakan kombinasi `useRef` untuk menampung masukan dan [`forwardRef`](/reference/react/forwardRef) untuk mengeksposnya ke komponen induk. Baca sebuah [panduan mendetail](/learn/manipulating-the-dom-with-refs#accessing-another-components-dom-nodes) di sini.
```

  // ...
```

Di sini, `playerRef` itu sendiri dapat bernilai null. Namun, Anda harus dapat meyakinkan pemeriksa tipe Anda bahwa tidak ada kasus di mana `getPlayer()` mengembalikan `null`. Kemudian gunakan `getPlayer()` di dalam event handler Anda.

</DeepDive>

---

## Pemecahan masalah {/*troubleshooting*/}

> ### Saya tidak bisa mendapatkan ref ke komponen khusus {/*i-cant-get-a-ref-to-a-custom-component*/}

Reviewer: ```suggestion
### Saya tidak bisa mendapatkan ref ke komponen kustom {/*i-cant-get-a-ref-to-a-custom-component*/}
```

Jika Anda mencoba mengoper `ref` ke komponen Anda sendiri seperti ini:

```js
const inputRef = useRef(null);

return <MyInput ref={inputRef} />;
```

Anda mungkin mendapatkan error di konsol:

<ConsoleBlock level="error">

Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?

</ConsoleBlock>

> Secara default, komponen Anda sendiri tidak mengekspos ref ke node DOM di dalamnya.

Reviewer: ```suggestion
Secara bawaan, komponen Anda sendiri tidak mengekspos ref ke node DOM di dalamnya.
```

Reviewer 2: @mazipan @resir014 This word needs to be confirmed in the [Glossary](https://github.com/reactjs/id.react.dev/wiki/Glosarium) later.

Discussion: https://github.com/reactjs/id.react.dev/discussions/337
